src/plugins/passwd/passwd.c

```diff
@@ -49,6 +49,79 @@ static int validatepwent (struct passwd * pwd)
 	return 0;
 }
 
+#if defined(USE_FGETPWENT_LOCAL)
+// clang-format off
+/* Taken from musl libc
+ *
+ * https://github.com/ifduyue/musl/blob/b4b1e10364c8737a632be61582e05a8d3acf5690/src/passwd/getpwent_a.c
+ * https://github.com/ifduyue/musl/blob/b4b1e10364c8737a632be61582e05a8d3acf5690/src/passwd/fgetpwent.c
+ */
+#include <pthread.h>
+
+static unsigned atou(char **s)
+{
+	unsigned x;
+	for (x=0; (unsigned char)(**s-'0')<10U; ++*s) x=10*x+((unsigned char)**s-'0');
+	return x;
+}
+
+int __getpwent_a(FILE *f, struct passwd *pw, char **line, size_t *size, struct passwd **res)
+{
+	ssize_t l;
+	char *s;
+	int rv = 0;
+	int cs;
+	pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
```

Comment:
find_package (Threads QUIET)
LINK_LIBRARIES ${CMAKE_THREAD_LIBS_INIT}
needed (see resolver)

Reply:
can we remove the pthread references all together? Does not seem worth it for the additional dependency.

Reply:
I did not look into what they do. But you have a guarantee that the same instance of a plugin is not called from multiple threads at the same time.